lib/land/trackers/api_tracker.rb

```diff
@@ -202,24 +229,63 @@ def user_agent
                      Land.config.blank_user_agent_string
 
         @user_agent = UserAgent[user_agent]
+        @user_agent.user_agent_type = UserAgentType['user']
 
         if Land.config.identify_crawlers && defined?(CrawlerDetect)
           crawler_detect = CrawlerDetect.new(user_agent)
-          user_agent_type = crawler_detect.is_crawler? ? 'crawl' : 'api'
-          @user_agent.update(user_agent_type: UserAgentType[user_agent_type])
+          @user_agent.user_agent_type = UserAgentType['crawl'] if crawler_detect.is_crawler?
         end
 
         browser = ::Browser.new(user_agent)
 
-        @user_agent.update(
-          browser: Browser[browser.name],
-          device: Device[browser.device.name],
-          platform: Platform[browser.platform.name],
-          browser_version: browser.version
-        )
+        update_device_resolution
+        update_browser_color_preferences(Browser[browser.name])
+
+        @user_agent.browser = Browser[browser.name]
+        @user_agent.device = Device[browser.device.name]
+        @user_agent.platform = Platform[browser.platform.name]
+        @user_agent.browser_version = browser.version
+        @user_agent.device_resolution = DeviceResolution[device_resolution]
+
+        if @user_agent.changed? && !@user_agent.save
+          error = @user_agent.errors.full_messages.join(', ')
+          log_error(error)
+        end
 
         @user_agent
       end
+
+      def update_device_resolution
+        return unless device_resolution.present? && (device_width || device_height || device_orientation)
+
+        resolution = DeviceResolution[device_resolution]
+        resolution.width = device_width
+        resolution.height = device_height
+        resolution.orientation = device_orientation
+
+        resolution.save if resolution.changed?
+      rescue StandardError => e
+        log_error("Error updating device resolution: #{e.message}")
+      end
+
+      def update_browser_color_preferences(browser)
+        return unless browser
+
+        browser.dark_mode = dark_mode
+        browser.light_mode = light_mode
+        browser.no_preference = no_preference
+
+        browser.save if browser.changed?
+      rescue StandardError => e
+        log_error("Error updating browser color preferences: #{e.message}")
+      end
+
+      def log_error(error)
```

Comment:
This is miss-named, and misused. Lets name it similar to our tagging method.

```suggestion
      def add_error_tag_to_land_span(error)
```